src/server/components/auth/express-session.ts

```diff
@@ -21,35 +21,39 @@ declare module "express-session" {
 }
 
 export async function configureExpressSession(server: Express): Promise<void> {
-  const secret = await getSecretValue(SECRET_NAME);
-
-  setTimeout(() => {
-    rotateSecret(SECRET_NAME).catch(noop);
-  }, random(200) * ONE_MINUTE);
-
-  const options: session.SessionOptions = {
-    secret: secret,
-    saveUninitialized: true,
-    resave: false,
-    proxy: !isLocalHost,
-    cookie: {
-      secure: !isLocalHost,
-      maxAge: isLocalHost ? ONE_DAY : FOUR_HOURS,
-    },
-    name: "lists_sid",
-  };
-
-  if (isRedisAvailable()) {
-    const redisClient = getRedisClient();
-    const RedisStore = connectRedis(session);
-
-    options.store = new RedisStore({
-      client: redisClient,
-      prefix: "lists_session_",
-    });
-
-    logger.info("Redis session storage initialized successfully");
+  try {
+    const secret = await getSecretValue(SECRET_NAME);
+
+    setTimeout(() => {
+      rotateSecret(SECRET_NAME).catch(noop);
+    }, random(200) * ONE_MINUTE);
+
+    const options: session.SessionOptions = {
+      secret: secret,
+      saveUninitialized: true,
+      resave: false,
+      proxy: !isLocalHost,
+      cookie: {
+        secure: !isLocalHost,
+        maxAge: isLocalHost ? ONE_DAY : FOUR_HOURS,
+      },
+      name: "lists_sid",
+    };
+
+    if (isRedisAvailable()) {
+      const redisClient = getRedisClient();
+      const RedisStore = connectRedis(session);
+
+      options.store = new RedisStore({
+        client: redisClient,
+        prefix: "lists_session_",
+      });
+
+      logger.info("Redis session storage initialized successfully");
+    }
+
+    server.use(session(options));
+  } catch (error) {
+    throw new Error(`configureExpressSession: Error ${(error as Error).message}`);
```

Comment:
sorry, I don't think my previous [comment](https://github.com/UKForeignOffice/lists/pull/476#discussion_r1005897349) was clear.  I meant to say the try-catch should be removed here and the error should bubble up. If the exception occurs on startup the app will crash anyway.